### PR TITLE
#485 add zlib to Exporter.export2thredds

### LINKED ISSUE #485: Allow zlib compression of output files for THREDDS
add `zlib=True` in `exporter.Exporter._copy_nc_var.createVariable`

### PR DESCRIPTION
**Pull request by akorosov**
Closes #485

### CODE REVIEW

**Comment by akorosov** [comment]
@korvinos , if you have time maybe you can review this PR?

File: nansat/exporter.py [hunk 1]
```diff
@@ -124,8 +124,16 @@ def export(self, filename='', bands=None, rm_metadata=None, add_geolocation=True
 
         self.logger.debug('Export - OK!')
 
-    def export2thredds(self, filename, bands={}, metadata=None, mask_name=None, no_mask_value=64, rm_metadata=None,
-                       time=None, created=None):
+    def export2thredds(self,
+        filename,
+        bands={},
```
Comment (korvinos) [inline nansat/exporter.py:129]: I do not really like the empty dictionary as an argument default. 
Maybe you could replace it with None, and check if bands is None in line 197

File: nansat/exporter.py [hunk 2]
```diff
@@ -254,7 +264,8 @@ def export2thredds(self, filename, bands={}, metadata=None, mask_name=None, no_m
         data.export(tmp_filename, rm_metadata=rm_metadata)
         del data
 
-        self._post_proc_thredds(tmp_filename, filename, bands, dstBands, time, global_metadata)
+        self._post_proc_thredds(
```
Comment (korvinos) [inline nansat/exporter.py:267]: It seems like a crucial last step inside the method, please add a comment line before that which gives a hint on what you are going for.

File: nansat/exporter.py [hunk 3]
```diff
@@ -124,8 +124,16 @@ def export(self, filename='', bands=None, rm_metadata=None, add_geolocation=True
 
         self.logger.debug('Export - OK!')
 
-    def export2thredds(self, filename, bands={}, metadata=None, mask_name=None, no_mask_value=64, rm_metadata=None,
-                       time=None, created=None):
+    def export2thredds(self,
```
Comment (korvinos) [inline nansat/exporter.py:127]: please refactor some variable names inside the function to satisfy pep8. For example `srcBands`

File: nansat/exporter.py [hunk 4]
```diff
@@ -300,8 +311,27 @@ def _create_dimensions(self, nc_inp, nc_out, time):
 
         return grid_mapping_name, grid_mapping_var_name
 
-    def _post_proc_thredds(self, tmp_filename, out_filename, bands, band_metadata, time, global_metadata):
-        """Post processing of file for THREDDS (add time variable and metadata)"""
+    def _post_proc_thredds(self,
```
Comment (korvinos) [inline nansat/exporter.py:314]: General comments for the function: 
* The function returns 0 which seems to be not needed
* Please refactor some variable names within the function according to the pep8

**Review by korvinos — CHANGES_REQUESTED** [review]
hi, looks good, I just added some minor comments on the style and commenting

File: nansat/exporter.py (hunk 1 above)
Comment (aperrin66) [reply nansat/exporter.py:129]: Sorry to show up uninvited, but I saw this in Slack and it's indeed a bad idea to use mutables as default arguments:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

File: nansat/exporter.py (hunk 3 above)
Comment (akorosov) [reply nansat/exporter.py:127]: Done

File: nansat/exporter.py (hunk 2 above)
Comment (akorosov) [reply nansat/exporter.py:268]: Done

File: nansat/exporter.py (hunk 4 above)
Comment (akorosov) [reply nansat/exporter.py:315]: Done

File: nansat/exporter.py (hunk 1 above)
Comment (akorosov) [reply nansat/exporter.py]: Absolutely agree 👍 
Done

**Comment by akorosov** [comment]
@korvinos, I'm done with your suggestions. Also increased test coverage a bit, even though it is not the point of this PR.

**Review by korvinos — APPROVED** [review]

**Pull request merged**